receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -73,7 +76,8 @@ func (r *pReceiver) Start(ctx context.Context, host component.Host) error {
 	if !r.cfg.UseStartTimeMetric {
 		jobsMap = internal.NewJobsMap(2 * time.Minute)
 	}
-	ocaStore := internal.NewOcaStore(ctx, r.consumer, r.logger, jobsMap, r.cfg.UseStartTimeMetric, r.cfg.StartTimeMetricRegex, r.cfg.Name())
+	receiverCtx := obsreport.ReceiverContext(ctx, r.cfg.Name(), transport)
```

Comment:
This needs to start from `context.Background()` because the context from Start should not be stored and used later (not you bug, but please fix that as well).

Reply:
Sorry, I was going to update this but it was merged. Do you need me to open another pr?